api/src/main/java/io/opencensus/trace/propagation/B3Format.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2017, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.trace.propagation;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import io.opencensus.trace.SpanContext;
+import java.text.ParseException;
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Implementation of the B3 propagation protocol. See
+ * <a href=https://github.com/openzipkin/b3-propagation>b3-propagation</a>.
+ *
+ * <p>This is only recommended to be used for backwards compatibility with systems like Zipkin.
+ * For new implementations we recommend using the {@code HttpFormat}.
+ *
+ * <p>Example of usage on the client:
+ *
+ * <pre>{@code
+ * private static final Tracer tracer = Tracing.getTracer();
+ * private static final B3Format b3Format = Tracing.getPropagationComponent().getB3Format();
+ * void onSendRequest() {
+ *   try (Scope ss = tracer.spanBuilder("Sent.MyRequest").startScopedSpan()) {
+ *     Map<HeaderName, String> b3Headers  =
+ *         b3Format.toHeaders(tracer.getCurrentContext().context());
+ *     // Send the request including the b3Headers and wait for the response.
+ *   }
+ * }
+ * }</pre>
+ *
+ * <p>Example of usage on the server:
+ *
+ * <pre>{@code
+ * private static final Tracer tracer = Tracing.getTracer();
+ * private static final B3Format b3Format = Tracing.getPropagationComponent().getB3Format();
+ * void onRequestReceived() {
+ *   // Get the B3 headers from the request.
+ *   Map<HeaderName, String> b3Headers = ...
+ *   SpanContext spanContext = b3Format.fromHeaders(b3Headers);
+ *   try (Scope ss =
+ *            tracer.spanBuilderWithRemoteParent("Recv.MyRequest", spanContext).startScopedSpan()) {
+ *     // Handle request and send response back.
+ *   }
+ * }
+ * }</pre>
+ */
+public abstract class B3Format {
+  public enum HeaderName {
```

Comment:
I think the name should include B3, such as `B3HeaderName`, since it can't represent all header names.

Reply:
possibly the intended use is B3Format.HeaderName.X (I use this a lot in other libraries)